pkg/asset/machines/azure/machines.go

```diff
@@ -110,10 +111,20 @@ func provider(platform *azure.Platform, mpool *azure.MachinePool, osImage string
 		image.SKU = mpool.OSImage.SKU
 		image.Version = mpool.OSImage.Version
 	} else {
-		imageID := fmt.Sprintf("/resourceGroups/%s/providers/Microsoft.Compute/images/%s", rg, clusterID)
+		// image gallery names cannot have dashes
+		galleryName := strings.Replace(clusterID, "-", "_", -1)
+		id := clusterID
 		if hyperVGen == "V2" {
-			imageID += "-gen2"
+			id += "-gen2"
 		}
+		imageVersion := rhcosVersion
+		if imageVersion != "" {
+			// image gallery image versions cannot have a "-" in the name and must be between 0-2147483647, so had to truncate the hour and minutes of the date
+			imageVersion = imageVersion[:len(imageVersion)-6]
+		} else {
+			imageVersion = "0.0.1"
```

Comment:
Is this supposed to work or should it be an error scenario?

Reply:
i didn't want to error out but just use a default version...but i don't think this will ever be empty so might be worth erroring out.

Reply:
the more i think about this, i don't think we need this default or to error out at all. the asset generation will error out if parsing fails and only other way it can be empty is if the json has no release string which i think we can safely assume it will. so i am not going to account for this. is that ok ?

Reply:
cleaned up by removing the unnecessary checks.